MantidPlot/src/Mantid/MantidPlotUtilities.cpp

```diff
@@ -0,0 +1,80 @@
+#include "MantidPlotUtilities.h"
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidGeometry/MDGeometry/IMDDimension.h"
+#include <MantidQtMantidWidgets/MantidDisplayBase.h>
+
+using namespace MantidQt::MantidWidgets;
+using Mantid::API::WorkspaceGroup_const_sptr;
+using Mantid::API::WorkspaceGroup_sptr;
+using Mantid::API::MatrixWorkspace_const_sptr;
+using Mantid::API::MatrixWorkspace_sptr;
+using Mantid::API::MatrixWorkspace;
+using Mantid::API::ExperimentInfo;
+using Mantid::HistogramData::Histogram;
+
+/**Compare two CurveSpecs to sort according to log value and
+* if equal by workspace index.
+* @param lhs left hand comparee
+* @param rhs right hand comparee
+* @returns true if right hand comparee has greater log value than left hand
+* comparee
+*/
+bool byLogValue(const CurveSpec &lhs, const CurveSpec &rhs) {
+  if (lhs.logVal == rhs.logVal)
+    return (lhs.index < rhs.index);
+  return (lhs.logVal < rhs.logVal);
+}
+
+/**
+ * Gets the given log value from the given workspace as a double.
+ * Should be a single-valued log!
+ * @param wsIndex :: [input] Index of workspace in group
+ * @param matrixWS :: [input] Workspace to find log from
+ * @param logName :: [input] Name of log
+ * @returns log value as a double, or workspace index
+ * @throws invalid_argument if log is wrong type or not present
+ */
+double getSingleWorkspaceLogValue(
+    size_t wsIndex, const Mantid::API::MatrixWorkspace_const_sptr &matrixWS,
+    const QString &logName) {
+  if (logName == MantidWSIndexWidget::WORKSPACE_INDEX || logName == "") {
+    return (double)wsIndex; // cast for plotting
```

Comment:
We also `return` here, so we can use an [early exit](http://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code) to remove the else statement and reduce code complexity